uaclient/contract.py

```diff
@@ -368,80 +369,87 @@ def _update_contract_machine(
 
         @return: Dict of the JSON response containing refreshed machine-token
         """
+        if not machine_id:
+            machine_id = system.get_machine_id(self.cfg)
+
         headers = self.headers()
         headers.update({"Authorization": "Bearer {}".format(machine_token)})
-        data = self._get_platform_data(machine_id)
-        data["activityInfo"] = self._get_activity_info()
+        data = {
+            "machineId": machine_id,
+            "activityInfo": self._get_activity_info(),
+        }
+        backcompat_data = _support_old_machine_info(data)
         url = API_V1_UPDATE_CONTRACT_MACHINE.format(
-            contract=contract_id, machine=data["machineId"]
+            contract=contract_id, machine=machine_id
         )
         response = self.request_url(
-            url, headers=headers, method="POST", data=data
+            url, headers=headers, method="POST", data=backcompat_data
         )
         if response.code != 200:
             raise exceptions.ContractAPIError(
                 url, response.code, response.body
             )
         if response.headers.get("expires"):
             response.json_dict["expires"] = response.headers["expires"]
-        machine_id = response.json_dict.get("machineTokenInfo", {}).get(
-            "machineId", data.get("machineId")
-        )
         return response.json_dict
 
-    def update_files_after_machine_token_update(
-        self, response: Dict[str, Any]
-    ):
-        self.cfg.machine_token_file.write(response)
-        system.get_machine_id.cache_clear()
-        data = self._get_platform_data(None)
-        machine_id = response.get("machineTokenInfo", {}).get(
-            "machineId", data.get("machineId")
-        )
-        self.cfg.write_cache("machine-id", machine_id)
-
-    def _get_platform_data(self, machine_id):
-        """Return a dict of platform-related data for contract requests"""
-        if not machine_id:
-            machine_id = system.get_machine_id(self.cfg)
-        return {
-            "machineId": machine_id,
+    def _get_activity_info(self):
```

Comment:
Why can't this function already returns the compatible structure we need to send to the Contract server ?

Reply:
It returns exactly this structure: https://github.com/canonical/ua-contracts/blob/develop/docs/contracts.yaml#L275

That is used directly as the body for `update_activity_token` and used directly as the value of "activityInfo" for the body of `update_contract_machine`.

Everything else is a special case that requires the same values but only a subset of them, or in add_contract_machine needs `lastAttachment` even though we're not attached yet.